docs(benchmarks): add Performance and Sizing Guide for record encryption (#3957)
kroxylicious/kroxylicious: kroxylicious-docs/docs/_modules/performance/ref-performance-limitations.adoc

////
  // Copyright Kroxylicious Authors.
  //
  // Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
////

:_mod-docs-content-type: REFERENCE

// file included in the following:
//
// performance-guide/index.adoc

[id='ref-performance-limitations-{context}']
= Measurement limitations

[role="_abstract"]
The following limitations apply to the benchmark results presented in this guide.

Single-pass measurements:: Each rate point was measured once.

> Environmental noise (pod scheduling, GC pauses, noisy neighbours) can affect individual probes.

Reviewer: ```suggestion
Environmental factors (pod scheduling, GC pauses, noisy neighbours) can affect individual probes.
```

Multi-pass aggregation would improve confidence intervals.

Single proxy pod:: All tests used a single proxy instance.

> Horizontal scaling characteristics are not yet measured.

Reviewer: ```suggestion
Horizontal scaling characteristics have not been measured.
```

> Single partition saturation:: The 1-topic rate sweep workload hits Kafka's per-partition throughput limit (~50,000 msg/s), which is close to the encryption ceiling.

Reviewer: ```suggestion
Single partition saturation:: The 1-topic rate sweep workload reaches Kafka's per-partition throughput limit (~50,000 msg/s), which is close to the measured encryption ceiling.
```

> The connection sweep and coefficient derivation used a 10-topic RF=1 workload to eliminate this ceiling.

Reviewer: ```suggestion
The connection sweep and coefficient derivation used a 10-topic workload with a replication factor of 1 to eliminate this ceiling.
```

> Hardware-specific:: Encryption is CPU-bound.

Reviewer: ```suggestion
Hardware-specific:: Encryption performance is CPU-bound.
```

> Results will differ on other CPU architectures and clock speeds.

Reviewer: ```suggestion
Results are expected to differ on other CPU architectures and clock speeds.
```

> The coefficient assumes AES-NI hardware acceleration.

Reviewer: ```suggestion
The coefficient assumes CPUs with AES-NI hardware acceleration.
```

> Linear scaling validated at 1, 2, and 4 cores:: The formula held across all three measurements.

Reviewer: ```suggestion
Linear scaling validated at 1, 2, and 4 cores:: The sizing formula held across all three measurements.
```

> Behaviour beyond 4 cores is untested but expected to follow the same model.

Reviewer: ```suggestion
Behavior beyond 4 cores is untested, but is expected to follow the same model.
```
